doc/common/breaking_changes.md

```diff
@@ -0,0 +1,104 @@
+This document contains a list of known breaking changes in the engineering system tools (mainly `eng/common` directories) and available workarounds. Typically this takes the form of infrastructure changes that are not backwards compatible with older versions of the engineering system configs/scripts/tools, mainly due to dependency changes in our test agent images.
+
+Table of Contents
+=================
+
+* [Credscan - You must install or update .NET to run this application.](#credscan---you-must-install-or-update-net-to-run-this-application)
+    * [Workaround](#workaround)
+* [fatal: specify directories rather than patterns](#fatal-specify-directories-rather-than-patterns)
+    * [Workaround](#workaround-1)
+
+
+## Credscan - You must install or update .NET to run this application.
+
+This error comes from the "CredScan running" step in the Compliance stage:
+
+```
+##[error]You must install or update .NET to run this application.
+```
+
+On 4/5/2023, dotnet 3.1 was removed from the 1es managed images used for Azure Pipelines runs. Older versions of credscan required dotnet 3.1.
+
+### Workaround
+
+Fix made and synced to all repositories from https://github.com/Azure/azure-sdk-tools/pull/5918
+
+To fix this issue, update your branch with the latest changes to `eng/common` in main, or cherry pick the following commit:
+
+```
+# azure-sdk-for-java
+git cherry-pick 2f9cdd9319f1867707b5872f12d97d5cdbbb077a
```

Comment:
I suspect you are being explicit to make this a simply copy and past but perhaps we just link to the tools PR and tell folks to find the sync merge. That would be more generic and isn't difficult to do.

Reply:
My main concern there is if you have to stack breaking changes on top of each other for an older branch, it could get pretty annoying to open up multiple sync PRs. It doesn't take me more than a minute to get all the cherry pick commits when I make a breaking change, since I already have the sync PRs open. But I can also see if we don't want the doc to get huge that it makes more sense to use the generic instruction.

Reply:
I'm not very worried about it at this point but something we might consider if we end up with more of these.

Reply:
I still don't think it's user friendly enough and it's going to create more work for us helping people. For example, I'm currently adding another breaking change referencing https://github.com/Azure/azure-sdk-tools/pull/5608. This PR has so many comments that the actual sync PR links are auto-hidden by the github UI. Then the user has to go and make sure to find the merge commit from the PR, not the actual PR commits. I think it's pretty likely we're going to have to help people troubleshoot either finding the right PR or the right commit, so it's going to be less effort to put the right commits into the markdown from the beginning. That way people can just click a link and copy a single command and there's less room for error.

Reply:
Fair. Perhaps we consider some collapsible sections to help reduce the clutter as this grows. https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections#creating-a-collapsed-section

Reply:
> For example, I'm currently adding another breaking change referencing #5608.

What did I break? 😖

Reply:
@konrad-jamrozik actually the opposite, you fixed something just in time (updating the codeowners tool version to the one built on net 6) because net 5 seems to have been removed from our build agents.

Reply:
So I guess the doc is somewhat misleading. It's not actually listing "Breaking Changes" but rather "Changes that are required for things to continue working." Not sure how I should title that (maybe "Known Backwards Compatibility Issues and Fixes" ???)

Reply:
Ok I updated the description to hopefully clarify what is meant by "breaking changes" and I added collapsible sections (nice find @weshaggard!).